fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/PantsTarget.scala

```diff
@@ -3,37 +3,48 @@ package scala.meta.internal.fastpass.pantsbuild
 import java.nio.file.Path
 import scala.meta.io.AbsolutePath
 
-case class PantsTarget(
-    name: String,
-    id: String,
-    dependencies: collection.Seq[String],
-    excludes: collection.Set[String],
-    platform: Option[String],
-    transitiveDependencies: collection.Seq[String],
-    compileLibraries: collection.Seq[String],
-    runtimeLibraries: collection.Seq[String],
-    isPantsTargetRoot: Boolean,
-    targetType: TargetType,
-    pantsTargetType: PantsTargetType,
-    globs: PantsGlobs,
-    roots: PantsRoots,
-    scalacOptions: List[String],
-    javacOptions: List[String],
-    extraJvmOptions: List[String],
-    directoryName: String,
-    classesDir: Path
+// NOTE(olafur): intentionally not a case class to use reference equality.
+class PantsTarget(
+    val name: String,
+    val id: String,
+    val dependencies: collection.Seq[String],
+    val javaSources: collection.Seq[String],
+    val excludes: collection.Set[String],
+    val platform: Option[String],
+    val libraries: collection.Seq[String],
+    val isPantsTargetRoot: Boolean,
+    val isPantsModulizable: Boolean,
+    val targetType: TargetType,
+    val pantsTargetType: PantsTargetType,
+    val globs: PantsGlobs,
+    val roots: PantsRoots,
+    val scalacOptions: List[String],
+    val javacOptions: List[String],
+    val extraJvmOptions: List[String],
+    val directoryName: String,
+    val baseDirectory: Path,
+    val classesDir: Path,
+    val internalSourcesJar: Path,
+    val strictDeps: Boolean,
+    val isSynthetic: Boolean,
+    val exports: Set[String],
+    val scope: PantsScope,
+    val targetBase: Option[String]
 ) {
+  require(
+    !classesDir.getFileName().toString().endsWith(".json"),
+    s"the classes directory '$classesDir' ends with '.json', " +
+      s"which Bloop will fail to load because it assumes it's a JSON file. " +
+      s"To fix this problem, we need to change the implementation in Fastpass " +
+      s"to not generate classes directories with the '.json' file extension."
+  )
```

Comment:
In what circumstances did you get a classes directory whose name ended in `.json`?

Reply:
It doesn't happen now because we generate random names for classes directories. It can happen if you use the Pants target `id` field as the directory name since many Pants targets are named "json"